Source/UI/NewUI/PaymentForm/ViewController/PaymentViewController.swift

```diff
@@ -297,3 +299,13 @@ extension PaymentViewController: SecurityCodeViewDelegate {
     delegate?.securityCodeIsUpdated(value: securityCode)
   }
 }
+
+extension PaymentViewController: CardNumberViewModelDelegate {
+    func update(cardNumber: String, scheme: Card.Scheme) {
+      securityCodeView.updateCardScheme(cardScheme: scheme)
+    }
+
+    func schemeUpdatedEagerly(to newScheme: Card.Scheme) {
+      securityCodeView.updateCardScheme(cardScheme: newScheme)
+    }
```

Comment:
what is the deffernce between  `func update` and `func schemeUpdatedEagerly` ?

Reply:
There are different times in the lifecycle of the card view. When the card view content is updated eagerly, it calls `schemeUpdatedEagerly` which only tells us the scheme it has so far (subject to change).

`update` is called when the editing has ended and returns the latest version of the card number present.

This allows us to sync the CVV validation once the card input is "final" (user dismissed keyboard), but also as the input is updated and scheme may keep changing from one key to another.

Reply:
I may suggest a new name for `update` to be `cardNumberDidFinishEditing` or any name related to description.

Reply:
I have looked now at making that change but it seems the wrong time to do it as its implications are wider.

To start with, `update(cardNumber:` doesn't strictly get called on did finish editing. If the editing is finished without a valid length, it will not be called at all. So the suggested syntax wouldn't make sense anymore... 
Additionally, once we handle the input (store for dispatch on button trigger) we will need to address the lifecycle better. For now this may get called if you introduce a full valid card number, but if you change it after to something invalid it will not be called. So the naming may better be decided on at that stage as the behaviour is reviewed in the full context.